lib/api/actions/functions.php

```diff
@@ -410,119 +410,189 @@ function beans_reset_action( $id ) {
 }
 
 /**
- * Get action.
+ * Get the action's configuration for the given ID and status. Returns `false` if the action is not registered with
+ * Beans.
  *
+ * @since  1.0.0
+ * @since  1.5.0 Made WPCS compliant.
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param string $status Status for which to get the action.
+ *
+ * @return array|bool
  */
 function _beans_get_action( $id, $status ) {
-
 	global $_beans_registered_actions;
 
-	$id = _beans_unique_action_id( $id );
+	$id                 = _beans_unique_action_id( $id );
+	$registered_actions = beans_get( $status, $_beans_registered_actions );
 
-	if ( ! $registered = beans_get( $status, $_beans_registered_actions ) ) {
+	// If the status is empty, return false, as no actions are registered.
+	if ( empty( $registered_actions ) ) {
 		return false;
 	}
 
-	if ( ! $action = beans_get( $id, $registered ) ) {
+	$action = beans_get( $id, $registered_actions );
+	// If the action is null, return false.
+	if ( is_null( $action ) ) {
 		return false;
 	}
 
 	return (array) json_decode( $action );
-
 }
 
 /**
- * Set action.
+ * Store the action's configuration for the given ID and status.
+ *
+ * What happens if the action's configuration is already registered?  If the `$overwrite` flag is set to `true`,
+ * then the new action's configuration is stored, overwriting the previous one. Else, the registered action's
+ * configuration is returned.
  *
+ * @since  1.0.0
+ * @since  1.5.0 Made WPCS compliant.
  * @ignore
+ * @access private
+ *
+ * @param string $id        A unique string used as a reference.
+ * @param array  $action    The action configuration to store.
+ * @param string $status    Status for which to store the action.
+ * @param bool   $overwrite Optional. When set to `true`, the new action's configuration is stored, overwriting a
+ *                          previously stored configuration (if one exists).
+ *
+ * @return array
  */
-function _beans_set_action( $id, $action, $status, $overwrite = false ) {
-
-	global $_beans_registered_actions;
-
+function _beans_set_action( $id, array $action, $status, $overwrite = false ) {
```

Comment:
I am getting an error when doing `beans_remove_action( 'beans_post_image' );` because the `$action` isn't an array in that case.
 
PS: I don't always run tests on PR so some bugs might be missed. Therefore we well have to run intense testing before the release as changes made to the API can have major impact.

Reply:
>I am getting an error when doing beans_remove_action( 'beans_post_image' ); because the $action isn't an array in that case.

As I'm working through the Actions API, I'm tightening up the tasks.  One of those is to only _set_ when we have the right format and information.  

I haven't got to the `beans_remove_action` function yet.  

But the overall strategy is this:

1. We should only store (set) an action that is properly formatted, meaning that it complies with our structure of:

- hook
- callback
- priority
- args

2. The `$action` then needs to be an array and have one or more of those keys within it.  Else, it's not an action that can be added or removed.

Reply:
I am happy with the strategy though we cannot merge breaking changes. So we must make sure the changes made in this PR doesn't break `beans_remove_action()` even if it means applying changes to it.

Reply:
Given that strategy, we'll do the following:

1. remove the change in this PR.
2. Re-add the change after we finish refactoring the Beans Actions API.